Recon/src/Recon/OverlayProducer.cxx

```diff
@@ -358,26 +360,26 @@ void OverlayProducer::onProcessStart() {
   }
 
   // replace by this line once the corresponding tweak to EventFile is ready:
-  //	overlayFile_ = std::make_unique<framework::EventFile>( overlayFileName_,
-  //true );
-  overlayFile_ = std::make_unique<framework::EventFile>(overlayFileName_);
+  overlayFile_ = std::make_unique<framework::EventFile>( overlayFileName_,true );
   overlayFile_->setupEvent(&overlayEvent_);
-
   // we update the iterator at the end of each event. so do this once here to
   // grab the first event in the processor
-  // TODO this could also be done N random times to get a randomness in which
-  // events get matched to what sim event. noticed that shifting by a fair chunk
-  // helps remove some weak but suspicious correlations between sim and overlay
-  // particle positions. leave it hardwired until we can reset the overlay event
-  // counter in nextEvent() (implemented with the EventFile definition above)
-  int nEventsShift_ = 23;
-  for (int iShift = 0; iShift < nEventsShift_; iShift++) {
+
+  TRandom2 * rd = new TRandom2( 0.); //RNSS undefined before produce()
+  // shift away a bit from the first event to avoid some weak correlations
+  // seen during initial valiadations
+  // TODO use the actual number of events in the file as upper limit
+  int startEvent = rd->Uniform(20., 500.); //assume we have at least 500 events
```

Comment:
I don't think you need to use the actual number of events in the file as the upper limit. I think you could just pick some absurbly high number (like 1M) as the upper limit because you already are wrapping around the file. It is a waste of time to go around the file again and again, but you only do it once at the start of a run.

Reply:
I'll test it a bit and see. I don't think an absurdly high number is needed though :) 10k is probably enough to cover the number of events in a pileup file. Still, I don't think the start event number needs to be pulled from a pool that is much larger than the actual times the file will be reused, and I don't think we'll want this number to be much larger than 100.